### ISSUE TITLE
LDAP Authentication Fails with usernames made up of all numbers

### ISSUE THREAD

**Issue by pfurches** [osTicket/osTicket-plugins#42]
OS Ticket 1.8 and 1.92
LDAP plugin 0.5  Using the PHAR install.

When a username is made up of all numbers, login fails with error access denied.  If you add any letter to the beginning of the username in AD and to the staff login, the scp login works.

I tested this with the same account.
The first time I used tk1234567890 and the login worked.
I changed the login in AD and OST to 1234567890 and it failed.

We have usernames with all numbers and this issue will prevent use from using OST as clients and staff will not be able to use SSO with our ticketing system now.

Any help please?

**Comment by ntozier** [comment]
I seem to recall a thread on the forums about this, and maybe even an issue being opened here on github.  You  might want to try searching to see what became of either.

**Comment by pfurches** [comment]
No joy.  The only article I found that mentioned LDAP and numbers was just a guess that another user confirmed was not the case.

**Comment by gitdesign** [comment]
I've only found it when the username has a leading 0 -- manual db mod fixes it though!

**Comment by pfurches** [comment]
Interesting.  All of our usernames are 10 digit numbers and none begin with 0.  But maybe it is related...?  What was your mod to work around this?  Can you share?

**Comment by gitdesign** [comment]
I have 8 digits all start with a leading 0

Ie 08259905

I jump into the DB directly and key it in.

Logan

**Comment by pfurches** [comment]
I am not having issues creating the accounts, they just cannot login to /scp.  If the username begins with a letter they login fine.  If it is a number it comes back access denied.

**Comment by greezybacon** [comment]
This is true. The authentication logic works as expected; however, the staff lookup will assume that something purely numeric is not an agent username, but an agent ID instead. The issue is in Staff::lookup inside osTicket core.
